File: src/main/java/com/itsci/mju/maebanjumpen/entity/PasswordUtil.java
package com.itsci.mju.maebanjumpen.entity;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

public class PasswordUtil {
    public static String hashPassword(String password) {
        if (password == null) {
            throw new IllegalArgumentException("Password cannot be null");
        }
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(password.getBytes(StandardCharsets.UTF_8));
            return Base64.getEncoder().encodeToString(hash);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException("Failed to hash password", e);
        }
    }

    public static boolean verifyPassword(String inputPassword, String storedHash) {
        if (inputPassword == null || storedHash == null) {
            return false;
        }
        String hashedInput = hashPassword(inputPassword);
        return hashedInput.equals(storedHash);
    }
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/PasswordUtil.kt
package com.itsci.mju.maebanjumpen.entity

import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.util.Base64

object PasswordUtil {
    fun hashPassword(password: String?): String {
        requireNotNull(password) { "Password cannot be null" }
        return try {
            val digest = MessageDigest.getInstance("SHA-256")
            val hash = digest.digest(password.toByteArray(StandardCharsets.UTF_8))
            Base64.getEncoder().encodeToString(hash)
        } catch (e: Exception) {
            throw RuntimeException("Failed to hash password", e)
        }
    }

    fun verifyPassword(inputPassword: String?, storedHash: String?): Boolean {
        if (inputPassword == null || storedHash == null) {
            return false
        }
        val hashedInput = hashPassword(inputPassword)
        return hashedInput == storedHash
    }
}

